.github/workflows/CI_build.yml

```diff
@@ -7,9 +7,79 @@ on:
 
 jobs:
 
-  analyze:
-    name: analyze / python-${{ matrix.python-version }} / ${{ matrix.os }}
+  first_check:
+    name: first code check / python-3.8 / ubuntu-latest
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8  
+      - name: Python info
+        run: |
+          which python
+          python --version
+      - name: Build package and create dev environment
+        run: |
+          python -m pip install --upgrade pip
+          python setup.py build
+          pip install -e .[dev]
+      - name: Test with coverage
+        run: |
+          pytest --cov --cov-report term --cov-report xml --junitxml=xunit-result.xml
+      - name: Correct coverage paths
+        run: sed -i "s+$PWD/++g" coverage.xml
```

Comment:
This is needed for SonarCloud scan. Move SonarCloud scan here too otherwise in build_pypi the SonarCloud will not have any coverage.

Reply:
Oh, good catch. I last-minute moved it away because I wanted to have a first CI run that does not rely on SonarCloud because that would not run on forks (say if people contribute PRs and first need to get used to the code standards). I think I found a better way not by only running SonarCloud if it is not forked.

Reply:
Nice find to exclude forks